src/main/java/com/minecolonies/rankup/util/AccountingUtils.java

```diff
@@ -0,0 +1,521 @@
+package com.minecolonies.rankup.util;
+
+import com.minecolonies.rankup.Rankup;
+import com.minecolonies.rankup.modules.core.config.AccountConfigData;
+import com.minecolonies.rankup.modules.databases.DatabaseModule;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.entity.living.player.User;
+import org.spongepowered.api.service.sql.SqlService;
+import org.spongepowered.api.service.user.UserStorageService;
+import uk.co.drnaylor.quickstart.exceptions.NoModuleException;
+
+import javax.sql.DataSource;
+import java.sql.Connection;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+/**
+ * Lots of accounts based utils (Used for database / file config adaption)
+ */
+public class AccountingUtils extends ConfigUtils
+{
+
+    private SqlService sql;
+    private Connection conn;
+
+    private              String table_id           = "player_stats";
+    private static final String UUID_COLUMN        = "UUID";
+    private static final String PLAYER_NAME_COLUMN = "PLAYER_NAME";
+    private static final String JOIN_DATE_COLUMN   = "JOIN_DATE";
+    private static final String LAST_JOIN_COLUMN   = "LAST_JOIN";
+    private static final String TIME_PLAYED_COLUMN = "TIME_PLAYED";
+
+    public AccountingUtils(final Rankup pl)
+    {
+        super(pl);
+    }
+
+    public void onInit()
+    {
+        getConn();
+    }
+
+    private DataSource getDataSource(String jdbcUrl) throws SQLException
+    {
+        table_id = getDatabasesConfig().sqlTablePrefix + "player_stats";
+        if (sql == null)
+        {
+            sql = Sponge.getServiceManager().provide(SqlService.class).get();
+        }
+        return sql.getDataSource(jdbcUrl);
+    }
+
+    private String getURI()
+    {
+        if (getDatabasesConfig().database.equalsIgnoreCase("mysql"))
+        {
+            return "jdbc:mysql://"
+                     + getDatabasesConfig().sqlUsername
+                     + ":"
+                     + getDatabasesConfig().sqlPassword
+                     + "@"
+                     + getDatabasesConfig().sqlAddress
+                     + "/"
+                     + getDatabasesConfig().sqlDatabase;
+        }
+        return "jdbc:h2:" + plugin.getConfigDir() + "/h2/playerstats";
+    }
+
+    private Connection getConn()
+    {
+        if (conn == null)
+        {
+            String uri = getURI();
+            try
+            {
+                conn = getDataSource(uri).getConnection();
+            }
+            catch (SQLException e)
+            {
+                e.printStackTrace();
+            }
+        }
+        return conn;
+    }
+
+    private ResultSet getQuery(final String query)
+    {
+        try
+        {
+            if (plugin.getModuleContainer().isModuleLoaded(DatabaseModule.ID) && conn != null)
+            {
+                final Statement stmt = conn.createStatement();
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Use try-with-resources or close this "Statement" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS2095)